src/routers/alpha-router/functions/best-swap-route.ts

```diff
@@ -465,12 +466,18 @@ export async function getBestSwapRouteBy(
     if (v2GasModel == undefined && v3GasModel == undefined) {
       throw new Error("Can't compute L1 gas fees.");
     } else {
+      // Now after v2 deploy everywhere, it's possible to to have v2 and v3 routes on L2 swap.
+      // So that we need to add both gas cost one by one
       const v2Routes = bestSwap.filter((routes) => routes.protocol === Protocol.V2);
       if (v2Routes.length > 0 && V2_SUPPORTED.includes(chainId)) {
         if (v2GasModel) {
           const v2GasCostL1 = await v2GasModel.calculateL1GasFees!(v2Routes as V2RouteWithValidQuote[]);
           gasCostsL1ToL2.gasUsedL1 = gasCostsL1ToL2.gasUsedL1.add(v2GasCostL1.gasUsedL1);
           gasCostsL1ToL2.gasUsedL1OnL2 = gasCostsL1ToL2.gasUsedL1OnL2.add(v2GasCostL1.gasUsedL1OnL2);
+          // It's possible that the gasCostL1USD currency is different from gasCostL1USD
```

Comment:
Sorry I still don't follow what this means here. Is this for when the usdGasToken on mainnet is different from the usdGasToken on the L2? And so in that case we just use the calculated L1 gas fees? The comment is specifying the same variable names so it's hard to follow

Reply:
I added more detailed explanations https://github.com/Uniswap/smart-order-router/pull/485/files#diff-bf0f39dc98ca22700a5c33efc44d6acf65a840bcface7d39297ffa99d20e9478R507-R521